api/v1beta1/rabbitmqcluster_types.go

```diff
@@ -100,6 +101,13 @@ type RabbitmqClusterSpec struct {
 	SecretBackend SecretBackend `json:"secretBackend,omitempty"`
 }
 
+type OverrideTrimmed struct {
+	// Override configuration for the RabbitMQ StatefulSet.
+	StatefulSet *runtime.RawExtension `json:"statefulSet,omitempty"`
```

Comment:
I like this approach in that it keeps our CRDs lean. If we ever decide to expose podTemplateSpec in any of our service operator specs I would like to consider a similar approach there.

Reply:
yes, its good to keep the CRD size small , but the downside is that it is unstructured/undocumented, especially if we want to only allow to customize/expose a subset of some templateSpec. so the usability for the user gets hard